R/France.R

```diff
@@ -0,0 +1,126 @@
+# nolint end
+#' @export
+#' @examples
+#' \dontrun{
+#' region <- France$new(verbose = TRUE, steps = TRUE, get = TRUE)
+#' region$return()
+#' }
+#'
+France <- R6::R6Class("France",
+  inherit = DataClass,
+  public = list(
+
+    # Core Attributes (amend each paramater for country specific infomation)
+    #' @field country name of country to fetch data for
+    country = "France",
+    #' @field supported_levels A list of supported levels.
+    supported_levels = list("1", "2"),
+    #' @field supported_region_names A list of region names in order of level.
+    supported_region_names = list("1" = "region", "2" = "department"),
+    #' @field supported_region_codes A list of region codes in order of level.
+    supported_region_codes = list("1" = "iso_3166_2", "2" = "ons_region_code"),
+    #' @field common_data_urls List of named links to raw data. This is empty
+    #' as for France data is divided into cases by level
+    common_data_urls = list(),
+    # nolint start
+    #' @field level_data_urls List of named links to raw data that are level
+    #' specific.
+    level_data_urls = list(
+      "1" = list("cases" = "https://www.data.gouv.fr/fr/datasets/r/001aca18-df6a-45c8-89e6-f82d689e6c01"),
+      "2" = list(
+        "cases" = "https://www.data.gouv.fr/fr/datasets/r/406c6a23-e283-4300-9484-54e78c8ae675",
+        "hosp" = "https://www.data.gouv.fr/fr/datasets/r/6fadff46-9efd-4c53-942a-54aca783c30c"
+      )
+    ),
+    # nolint end
+    #' @field source_data_cols existing columns within the raw data
+    source_data_cols = c("cases_new", "tested_new"),
+
+    #' @description Set up a table of region codes for clean data
+    #' @importFrom tibble tibble
+    #' @importFrom dplyr select
+    set_region_codes = function() {
+      if (self$level == "1") {
+        self$codes_lookup$`1` <- france_codes %>%
+          select(
+            .data$level_1_region_code,
+            .data$level_1_region,
+            .data$insee_code
+          )
+      } else if (self$level == "2") {
+        self$codes_lookup$`2` <- france_codes
+      }
+    },
+
+    #' @description France Specific Region Level Data Cleaning
+    #'
+    #' @importFrom dplyr filter mutate left_join select
+    #' @importFrom lubridate as_date ymd
+    clean_level_1 = function() {
+      self$data$clean <- self$data$raw$cases %>%
+        filter(cl_age90 == 0) %>%
+        select(
+          date = jour,
+          insee_code = reg,
+          cases_new = P,
+          tested_new = `T`
+        ) %>%
+        mutate(date = as_date(ymd(date))) %>%
+        left_join(
+          self$codes_lookup$`1`,
+          by = c("insee_code")
+        )
+    },
+
+    #' @description France Specific Department Level Data Cleaning
+    #'
+    #' @importFrom dplyr filter mutate left_join rename select full_join
+    #' @importFrom lubridate as_date ymd
+    clean_level_2 = function() {
+      cases_data <- self$data$raw$cases %>%
+        filter(cl_age90 == 0) %>%
```

Comment:
some of this looks like it could be common cleaning?

Reply:
The column selection is different:

`level_2_region_code = dep,` vs `insee_code = reg,`

Is there a nice way of coding these in with "self$level" in a `clean_common`?

I tried storing them in a named list but getting them out and setting was ugly!

Reply:
some of the selects and joins are different: `insee_code = reg`, `level_2_region_code = dep`

Reply:
I really meant just the first filter 😆 so no biggy